sktime/tests/test_all_estimators.py

```diff
@@ -54,6 +55,61 @@
 )
 
 
+generator_dict = dict()
+
+
+def is_excluded(test_name, est):
+    """Shorthand to check whether test test_name is excluded for estimator est."""
+    return test_name in EXCLUDED_TESTS.get(est.__name__, [])
+
+
+def _generate_estimator_class(test_name, **kwargs):
+    """Return estimator class fixtures.
+
+    Fixtures parameterized
+    ----------------------
+    estimator_class: estimator inheriting from BaseObject
+        ranges over all estimator classes not excluded by EXCLUDED_TESTS
+    """
+    estimator_classes_to_test = [
+        est for est in ALL_ESTIMATORS if not is_excluded(test_name, est)
+    ]
+    estimator_names = [est.__name__ for est in estimator_classes_to_test]
+
+    return estimator_classes_to_test, estimator_names
+
+
+generator_dict["estimator_class"] = _generate_estimator_class
+
+
+def _generate_estimator_instance(test_name, **kwargs):
+    """Return estimator instance fixtures.
+
+    Fixtures parameterized
+    ----------------------
+    estimator_instance: instance of estimator inheriting from BaseObject
+        ranges over all estimator classes not excluded by EXCLUDED_TESTS
+        instances are generated by create_test_instance class method
+    """
+    # we need to exclude instances from classes on the exclude list
+    estimator_classes_to_test = [
```

Comment:
this seems duplicated from `_generate_estimator_class` wouldn't it be more convenient o call `_generate_estimator_class` instead? I would also allow to not have `is_excluded` as a separate function since it would be absorbed by `_generate_estimator_class` as a statement.

Reply:
similar - we need to fill the abstract hole for generating the `estimator_instance` by providing a function `_generate_estimator_instance`. I'd like to have `is_excluded` separate, too, since it wraps the constant we may want to move, or replace by alternative logic.

Reply:
apologies, I misunderstood this statement - I thought you suggested to *replace* `_generate_estimator_instance`.
Calling `_generate_estimator_class` is indeed better here.